### PR TITLE
Upgrade node-addon-api Object Wrap example for multi-context support

### PR DESCRIPTION
**Pull request by anfilat**
It replaced global constructor with new SetInstanceData\GetInstanceData API.

### CODE REVIEW

File: 6_object_wrap/node-addon-api/package.json [hunk 1]
```diff
@@ -5,8 +5,11 @@
   "main": "addon.js",
   "private": true,
   "gypfile": true,
+  "engines": {
+    "node": ">= 14.0.0"
+  },
```
Comment (mhdawson) [inline 6_object_wrap/node-addon-api/package.json:10]: I don't think we'd want to add this. If the example needs to be different for multi-context and that is only available in later Node.js versions, then we should add a new example versus changing the old one. Ideally we would have an example that would work across all Node.js LTS versions.

Comment (anfilat) [reply 6_object_wrap/node-addon-api/package.json]: Ok, I changed it and it a new example now. But there is linked bug  https://github.com/nodejs/node-addon-api/issues/730

**Comment by jschlight** [comment]
@anfilat There is a newer example of using `ObjectWrap` on this same examples repository that does not involve a static constructor function:

https://github.com/nodejs/node-addon-examples/tree/master/object-wrap-demo/node-addon-api

As far as I can tell, this newer example is context aware. Will you be able to meet your objectives by following the pattern of this newer example?

A question for the N-API team. In what cases is a static constructor function required? Or could it be the case that the [6_object_wrap](https://github.com/nodejs/node-addon-examples/tree/master/6_object_wrap) example was simply a direct translation from NAN and no longer a good example to be promoting?

**Comment by anfilat** [comment]
@jschlight As I see, there is a similar PR from the N-API team member - https://github.com/nodejs/node-addon-examples/pull/139

**Comment by jschlight** [comment]
@anfilat Thank you. I will followup with the N-API team.

File: multiple_load_object_wrap/node-addon-api/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "object_wrap",
+  "version": "0.0.0",
+  "description": "Multiple load example of ObjectWrap",
+  "main": "addon.js",
+  "private": true,
+  "gypfile": true,
+  "engines": {
+    "node": ">= 14.0.0"
+  },
```
Comment (gabrielschulhof) [inline multiple_load_object_wrap/node-addon-api/package.json:10]: This needs to work with all supported versions of Node.js.

Comment (gabrielschulhof) [reply multiple_load_object_wrap/node-addon-api/package.json:10]: @anfilat using the value for "engiens" as suggested at https://github.com/nodejs/node-addon-examples/pull/139#issuecomment-637896592 should make the CI green, and it should actually test your code on Travis at least, even if it doesn't test it on GitHub actions.

Comment (gabrielschulhof) [reply multiple_load_object_wrap/node-addon-api/package.json:10]: Oh, wait, #139 needs to land to update the .travis.yml removing the non-supported Node.js 8 first 🙂

Comment (anfilat) [reply multiple_load_object_wrap/node-addon-api/package.json:10]: @gabrielschulhof I think I should close this PR. It's duplicate with https://github.com/nodejs/node-addon-examples/pull/139 now

**Comment by qxred** [comment]
Seems this issues exists when using cmake-js (6.1.0) instead of gyp , can anyone help?

**Pull request closed without merging**